code/modules/power/supermatter/supermatter.dm

```diff
@@ -187,6 +187,12 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 
 	/// cooldown tracker for accent sounds,
 	var/last_accent_sound = 0
+	///Var that increases from 0 to 1 when a psycologist is nearby, and decreases in the same way
+	var/psyCoeff = 0
+	///A pinkish overlay used to denote the presance of a psycologist. We fade in and out of this depending on the amount of time they've spent near the crystal
+	var/obj/overlay/psy/psyOverlay = new /obj/overlay/psy/
+	///A list of active overlays, we use this to update .overlays more cleanly
+	var/active_overlays = list()
```

Comment:
Turn this into a lazy list.

Reply:
Can do.

Reply:
I uh... I'm not quite sure what you mean.

Reply:
Ask in the coder channel what a lazy list is, feel free to @ me.